### PR TITLE
fix: improve non existing project errors

### LINKED ISSUE #3152: Show the full 404 page when a project/group is not found/visible to the user
Currently, the missing projects and groups show an error

image

The page should be the 404 currently used for missing v1 projects, similar to https://dev.renku.ch/projects/foo/bar.

### PR DESCRIPTION
**Pull request by lorenzo-cavazzi**
Shows a properly formatted error as it comes from the backend API.
The same style has been applied to the group page.

image

/deploy
fix #3152

### CODE REVIEW

**Review by leafty — CHANGES_REQUESTED** [review]
We should show a full page 404. I can have a look at doing it, since it requires routing changes.

**Comment by lorenzo-cavazzi** [comment]
> We should show a full page 404. I can have a look at doing it, since it requires routing changes.

Fine with me :+1: 
If you find the time in this cooldown, we can drop this PR. Otherwise, this PR still improves the current status and we can probably merge it as an intermediate step.

File: client/src/features/ProjectPageV2/ProjectPageContainer/ProjectPageContainer.tsx [hunk 1]
```diff
@@ -48,36 +47,26 @@ function ProjectPageContainerInner({
...
-          <p>Could not retrieve project</p>
-          <RtkErrorAlert error={error} />
+          {error ? (
+            <RtkOrNotebooksError error={error} />
+          ) : (
+            <p>Could not retrieve the project.</p>
+          )}
+          <p>
+            Click here to{" "}
+            <Link to={Url.get(Url.pages.projectV2.list)}>
```
Comment (leafty) [inline client/src/features/ProjectPageV2/ProjectPageContainer/ProjectPageContainer.tsx:61]: Can we avoid `Url.js`?

**Review by leafty — APPROVED** [review]

File: client/src/features/ProjectPageV2/ProjectPageContainer/ProjectPageContainer.tsx (hunk 1 above)
Comment (lorenzo-cavazzi) [reply client/src/features/ProjectPageV2/ProjectPageContainer/ProjectPageContainer.tsx:61]: Yes! I'll take the occasion to remove it from a few other V2 places. I'll push it to another PR

**Pull request merged**